cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1389,7 +1389,33 @@
 ;; immediate as an immediate to the instruction itself.
 (rule (vec_insert_lane $F64X2 vec val 1) (movlhps vec val))
 
-;;;; Rules for `imax` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;; Rules for `imin`, `imax`, `umin`, `umax` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; `i64` and smaller.
+
+(decl cmp_and_choose (Type CC Value Value) ValueRegs)
+(rule (cmp_and_choose (fits_in_64 ty) cc x y)
+      (let ((x_reg Reg (put_in_reg x))
+            (y_reg Reg (put_in_reg y))
+            (size OperandSize (raw_operand_size_of_type ty)))
+           (value_reg (with_flags_1 (cmp size (RegMemImm.Reg x_reg) y_reg)
+                                    (cmove ty cc (RegMem.Reg y_reg) x_reg)))))
+
+(rule (lower (has_type (fits_in_64 ty) (umin x y)))
+      (cmp_and_choose ty (CC.B) x y))
+
+(rule (lower (has_type (fits_in_64 ty) (umax x y)))
+      (cmp_and_choose ty (CC.NB) x y))
+
+(rule (lower (has_type (fits_in_64 ty) (imin x y)))
+      (cmp_and_choose ty (CC.L) x y))
+
+(rule (lower (has_type (fits_in_64 ty) (imax x y)))
+      (cmp_and_choose ty (CC.NL) x y))
+
+;; Wide
+
+; `imax`
```

Comment:
Just for consistency with the rest of the file, could you use

```lisp
;; SSE.
```

instead of "wide" and use `;;` instead of `;` comments? Thanks!